pkg/agent/cache/apiservice.go

```diff
@@ -0,0 +1,137 @@
+package cache
+
+import (
+	"fmt"
+
+	defs "github.com/Axway/agent-sdk/pkg/apic/definitions"
+
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	"github.com/Axway/agent-sdk/pkg/cache"
+	"github.com/Axway/agent-sdk/pkg/util"
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+// API service cache management
+
+// AddAPIService - add/update APIService resource in cache
+func (c *cacheManager) AddAPIService(svc *v1.ResourceInstance) error {
+	apiID, err := util.GetAgentDetailsValue(svc, defs.AttrExternalAPIID)
+	if err != nil {
+		return fmt.Errorf("failed to get external API ID from APIService resource: %s", err)
+	}
+	if apiID != "" {
+		defer c.setCacheUpdated(true)
+		apiName, _ := util.GetAgentDetailsValue(svc, defs.AttrExternalAPIName)
```

Comment:
you always guaranteed an apiName?  probably?

Reply:
not sure, not my code, just split it to multiple files

Reply:
Ok..  just wondering in case since we aren't check for "" or nil.